sklearn_numba_dpex/common/random.py

```diff
@@ -0,0 +1,307 @@
+    splitmix64_rshift_1 = uint32(30)
+    splitmix64_rshift_2 = uint32(27)
+    splitmix64_rshift_3 = uint32(31)
+
+    @dpex.func
+    def _splitmix64_next(state):
+        new_state = z = state + splitmix64_const_1
+        z = (z ^ (z >> splitmix64_rshift_1)) * splitmix64_const_2
+        z = (z ^ (z >> splitmix64_rshift_2)) * splitmix64_const_3
+        return new_state, z ^ (z >> splitmix64_rshift_3)
+
+    jump_const_1 = uint64(0x2BD7A6A6E99C2DDC)
+    jump_const_2 = uint64(0x0992CCAF6A6FCA05)
+    jump_const_3 = uint64(1)
+    jump_init = uint64(0)
+    long_2 = int64(2)
+    long_64 = int64(64)
+
+    @dpex.func
+    def _xoroshiro128pp_jump(states, index):
+        """Advance the RNG in ``states[index]`` by 2**64 steps."""
+        s0 = jump_init
+        s1 = jump_init
+
+        for i in range(long_2):
+            if i == zero_idx:
+                jump_const = jump_const_1
+            else:
+                jump_const = jump_const_2
+            for b in range(long_64):
+                if jump_const & jump_const_3 << uint32(b):
+                    s0 ^= states[index, zero_idx]
+                    s1 ^= states[index, one_idx]
+                _xoroshiro128pp_next(states, index)
+
+        states[index, zero_idx] = s0
+        states[index, one_idx] = s1
+
+    # TODO: it seems that the reference python implementation in the package
+    # `randomgen` that inspired this code contains errors.
+    # There's an issue here: https://github.com/bashtage/randomgen/issues/321
+    # This website https://prng.di.unimi.it/ seems to provide a ground truth
+    # implementation at https://prng.di.unimi.it/xoroshiro128plusplus.c but is not
+    # interfaced with python.
+    # We choose to mimic the reference implementation given in C even if there's no
+    # easy, reproducible set of instructions to test that it's working as expected.
+    # If the `randomgen` current implementation proves to be good, then the following
+    # block should be used instead, and the tests should be adapted. Else, the block
+    # can be removed.
+
+    # jump_const_1 = uint64(0xDF900294D8F554A5)
+    # jump_const_2 = uint64(0x170865DF4B3201FC)
+    # jump_const_3 = uint64(1)
+    # jump_init = uint64(0)
+    # long_2 = int64(2)
+    # long_64 = int64(64)
+
+    # @dpex.func
+    # def _xoroshiro128pp_jump(states, index):
+    #     """Advance the RNG in ``states[index]`` by 2**64 steps."""
+    #     s0 = jump_init
+    #     s1 = jump_init
+
+    #     for i in range(long_2):
+    #         if i == zero_idx:
+    #             jump_const = jump_const_1
+    #         else:
+    #             jump_const = jump_const_2
+    #         for b in range(long_64):
+    #             if jump_const & jump_const_3 << uint32(b):
+    #                 s0 ^= states[index, zero_idx]
+    #                 s1 ^= states[index, one_idx]
+    #             _xoroshiro128p_next(states, index)
+
+    #     states[index, zero_idx] = s0
+    #     states[index, one_idx] = s1
```

Comment:
It looks like this part of the comment is duplicated and can be removed.

```suggestion
```

Reply:
This was not entirely identical: `_xoroshiro128p_next` is called in this commented code, whereas `_xoroshiro128pp_next` is called, as indicated by @fcharras in a comment which was integrated in the meantime.